hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/curator/ZKCuratorManager.java

```diff
@@ -157,12 +175,44 @@ public void start(List<AuthInfo> authInfos) throws IOException {
       authInfos.add(new AuthInfo(zkAuth.getScheme(), zkAuth.getAuth()));
     }
 
+    /* Pre-check on SSL/TLS client connection requirements to emit the name of the
+    configuration missing. It improves supportability. */
+    if(sslEnabled) {
+      if (StringUtils.isEmpty(conf.get(CommonConfigurationKeys.ZK_SSL_KEYSTORE_LOCATION))) {
+        throw new ConfigurationException(
```

Comment:
Nit: Can you extract this to a helper method?
Only the config name should be passed, the rest of the method body (even the exception message) can be the same.

Reply:
I meant a method called something like validateX(String confKey) that throws the exception. The exception message is repeated 4 times, but it's not the end of the world if we don't do this. It's okay how it is now :)